apache/zeppelin
# [ZEPPELIN-3092] GitHub Integration

### What is this PR for?
GitHub integration as a storage for notebooks.


### What type of PR is it?
Feature

### What is the Jira issue?
[ZEPPELIN-3092](https://issues.apache.org/jira/browse/ZEPPELIN-3092)

### How should this be tested?
1. Change the configuration in `zeppelin-site.xml` to enable GitHub integration (add GitHub url, username, access token and origin) as described in https://github.com/apache/zeppelin/compare/master...mohamagdy:zeppelin-3092-remote-github-integration?expand=1#diff-89104d48f0358450399a6f679bba9c4f
2. Start the Zeppelin server
3. Open an existing notebook or create a new notebook
4. Do some changes to the notebook, for example add a new paragraph
5. Click on the versioning button on the top menu to commit and save changes
6. Checkout the changes in the GitHub repository. The changes should be reflected 

### Questions:
* **Does the licenses files need update?**
No

* **Is there breaking changes for older versions?**
No

* **Does this needs documentation?**
Yes. Documentation is updated as part of the pull request.
